cardinal/world.go

```diff
@@ -342,11 +433,11 @@ func (w *World) Shutdown() error {
 	if w.cleanup != nil {
 		w.cleanup()
 	}
-	ok := w.gameSequenceStage.CompareAndSwap(gamestage.StageRunning, gamestage.StageShuttingDown)
-	if !ok {
-		// Either the world hasn't been started, or we've already shut down.
-		return nil
-	}
+	// ok := w.gameSequenceStage.CompareAndSwap(gamestage.StageRunning, gamestage.StageShuttingDown)
+	// if !ok {
+	//	// Either the world hasn't been started, or we've already shut down.
+	//	return nil
+	// }
```

Comment:
dead code?

Reply:
i uncommented this and also just use store and compare and swap because there is no guarantee that the shutdown is only triggered after the game is at the running stage.